src/formatDate.js

```diff
@@ -50,7 +50,50 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldDate = date.split(fromFormat[3]);
+  const monthIndex = fromFormat.indexOf('MM');
+  const dateIndex = fromFormat.indexOf('DD');
+
+  let yearIndex = null;
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i] === 'YY' || fromFormat[i] === 'YYYY') {
+      yearIndex = i;
+    }
+  }
+
+  const newDate = oldDate[dateIndex];
+  const newMonth = oldDate[monthIndex];
+  let newYear = oldDate[yearIndex];
+
+  const newMonthIndex = toFormat.indexOf('MM');
+  const newDateIndex = toFormat.indexOf('DD');
+
+  const result = [];
+
+  result.length = 3;
```

Comment:
можно еще так записать:
```suggestion
  const result = new Array(3);
```